### PR TITLE
fix: connectClient freezes on request to dead server

### PR DESCRIPTION
**Pull request by denis-tingaikin**
Signed-off-by: Denis Tingaikin <denis.tingajkin@xored.com>

Closes https://github.com/networkservicemesh/deployments-k8s/pull/102

### CODE REVIEW

File: pkg/networkservice/common/connect/client.go [hunk 1]
```diff
@@ -34,7 +34,11 @@ type connectClient struct {
 }
 
 func (c *connectClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	return next.Client(ctx).Request(ctx, request, opts...)
+	resp, err := next.Client(ctx).Request(ctx, request, opts...)
+	if err != nil {
```
Comment (Bolodya1997) [inline pkg/networkservice/common/connect/client.go:38]: It doesn't mean that there is something wrong with server, it can be something wrong with request.

Comment (Bolodya1997) [reply pkg/networkservice/common/connect/client.go:38]: Or it can be something wrong with someone (not next host) else in chain.

Comment (denis-tingaikin) [reply pkg/networkservice/common/connect/client.go]: Yep, it is also is covered by TestNSMGR_SelectsRestartingEndpoint :)  
Fixed.

File: pkg/networkservice/common/connect/client.go [hunk 2]
```diff
@@ -1,6 +1,6 @@
-// Copyright (c) 2020 Doc.ai and/or its affiliates.
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
 //
-// Copyright (c) 2020 Cisco and/or its affiliates.
+// Copyright (c) 2020-2021 Cisco and/or its affiliates.
```
Comment (artem-belov) [inline pkg/networkservice/common/connect/client.go:3]:
```
// Copyright (c) 2020 Cisco and/or its affiliates.
//
// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
```
Also has to work and seems better

**Review by Bolodya1997 — APPROVED** [review]

**Review by haiodo — APPROVED** [review]

**Pull request merged**